File: dynamic_listings/python/default/AI/euclidean_distance.py
from math import sqrt


class Tile:
    def __init__(self, x: int, y: int) -> None:
        self.x: int = x
        self.y: int = y


def euclidean_distance(start: Vector2, goal: Vector2) -> float:
    return sqrt((start.x - goal.x) ** 2 + (start.y - goal.y) ** 2)

File: dynamic_listings/rust/default/AI/euclidean_distance.rs
struct Tile {
    x: u32,
    y: u32,
}

fn euclidean_distance(start: &Tile, goal: &Tile) -> f32 {
    ((start.x - goal.x).pow(2) as f32 + (start.y - goal.y).pow(2) as f32).sqrt()
}
